astropy/tests/helper.py

```diff
@@ -192,11 +200,82 @@ def assert_quantity_allclose(actual, desired, rtol=1.0e-7, atol=None, **kwargs):
     This is a :class:`~astropy.units.Quantity`-aware version of
     :func:`numpy.testing.assert_allclose`.
     """
-    import numpy as np
-
     from astropy.units.quantity import _unquantify_allclose_arguments
 
     __tracebackhide__ = True
     np.testing.assert_allclose(
         *_unquantify_allclose_arguments(actual, desired, rtol, atol), **kwargs
     )
+
+
+def assert_objects_equal(
+    actual: AstropyArrayLike,
+    desired: AstropyArrayLike,
+    /,
+    *,
+    attrs: list[str] | None = None,
+    compare_class: bool = True,
+) -> None:
+    """Compare objects by their shape and attributes, including those provided by .info
+
+    Parameters
+    ----------
+    actual and desired: AstropyArrayLike (positional only)
+        these objects must have attributes `shape` and `info`.
+        Typically this means astropy ndarray subclasses or
+        :class:`astropy.utils.shapes.ShapedLikeNDArray`.
+
+    attrs: list[str] (optional, keyword only)
+        a list of attributes to compare in addition to the default list:
+        ['info.name', 'info.format', 'info.unit',
+        'info.description', 'info.meta', 'info.dtype']
+
+    compare_class: bool (default: False, keyword only)
+        Whether to check that the two objects are instances of the same class
+
+    Raises
+    ------
+    AssertionError
+        If actual doesn't compare equal to desired.
+    """
+    if compare_class:
+        assert actual.__class__ is desired.__class__
+
+    assert actual.shape == desired.shape
+
+    info_attrs = [
+        "info.name",
+        "info.format",
+        "info.unit",
+        "info.description",
+        "info.meta",
+        "info.dtype",
+    ]
+    for attr in info_attrs + (attrs or []):
+        a1 = actual
+        a2 = desired
+        for subattr in attr.split("."):
+            try:
+                a1 = getattr(a1, subattr)
+                a2 = getattr(a2, subattr)
+            except AttributeError:
+                a1 = a1[subattr]
+                a2 = a2[subattr]
+
+        # Mixin info.meta can None instead of empty OrderedDict(), #6720 would
```

Comment:
The pull request referenced here is closed. Have you checked if the comment is still relevant?

Reply:
yes, it is.